dom.bs

```diff
@@ -177,19 +177,59 @@ first <a>following</a> <a for=tree>sibling</a> or null if it has no <a for=tree>
 added.
 
 
-<h3 id=namespaces>Namespaces</h3>
+<h3 id=namespaces>Name validation</h3>
 
-<p>To <dfn export>validate</dfn> a <var>qualifiedName</var>, <a>throw</a> an
-"{{InvalidCharacterError!!exception}}" {{DOMException}} if <var>qualifiedName</var> does not match
-the <code><a type>QName</a></code> production.
+<p>A [=string=] is a <dfn>valid namespace prefix</dfn> if its [=string/length=] is at least 1 and it
+does not contain [=ASCII whitespace=], U+0000 NULL, U+002F (/), or U+003E (>).
 
-<p>To <dfn export>validate and extract</dfn> a <var>namespace</var> and <var>qualifiedName</var>,
-run these steps:
+<p>A [=string=] is a <dfn>valid attribute local name</dfn> if its [=string/length=] is at least 1
+and it does not contain [=ASCII whitespace=], U+0000 NULL, U+002F (/), U+003D (=), or U+003E (>).
+
+<p>A [=string=] |name| is a <dfn>valid element local name</dfn> if the following steps return true:
 
 <ol>
- <li><p>If <var>namespace</var> is the empty string, then set it to null.
+ <li><p>If |name|'s [=string/length=] is 0, then return false.
 
- <li><p><a>Validate</a> <var>qualifiedName</var>.
+ <li>
+  <p>If |name|'s 0th [=code point=] is an [=ASCII alpha=], then:
+
+  <ol>
+   <li><p>If |name| contains [=ASCII whitespace=], U+0000 NULL, U+002F (/), or U+003E (>), then
+   return false.
+
+   <li><p>Return true.
```

Comment:
So does this mean that characters past U+10FFFF are allowed? Like `a\u110000`?

Reply:
I guess this might not be testable since `String.fromCodePoint(0x11000)` throws an exception, so maybe it doesn't matter?

Reply:
> So does this mean that characters past U+10FFFF are allowed?

No. https://infra.spec.whatwg.org/#code-points limits code points to U+10FFFF.

Reply:
Yeah, technically such things are not strings. It should be impossible for bytes above 0x10FFFF to appear inside the C++ string data structures that you're processing. You can't get them in their via either JavaScript APIs or the HTML parsing.

Reply:
Yeah, it seems unlikely you'd need to hold a uint32 to implement this. (ICU has some suitable asserts if for some reason you do.)

Reply:
Awesome, thanks! I was dealing with a C++ test that was manually setting the memory to funny things, sounds like this spec is good as-is.